integreat_cms/cms/utils/repair_tree.py

```diff
@@ -0,0 +1,259 @@
+    to fix hierarchy and sorts siblings by (potentially inconsistent) lft.
+    """
+
+    def __init__(self):
+        """
+        Creates a fixed tree when initializing class but does not save results
+        """
+        self.broken_nodes = list(Page.objects.all().order_by('tree_id', 'lft'))
+        # Dictionaries keep the insert order as of Python 3.7
+        self.fixed_nodes = {}
+        self.fix_root_nodes()
+        self.fix_child_nodes()
+
+    def fix_root_nodes(self):
+        """
+        extract root nodes and reset lft + rgt values
+        """
+        tree_counter = 1
+        for node in self.broken_nodes:
+            if not node.parent_id:
+                node.lft = 1
+                node.rgt = 2
+                node.depth = 1
+                node.fixed_children = []
+                node.tree_id = tree_counter
+                tree_counter = tree_counter + 1
+                self.fixed_nodes[node.pk] = node
+
+    def fix_child_nodes(self):
+        """
+        Get all remaining (child) nodes, add add them to the new/fixed tree
+        """
+        for node in self.broken_nodes:
+            if node.parent_id:
+                parent = self.fixed_nodes[node.parent_id]
+                node.fixed_children = []
+                node = self.calculate_lft_rgt(node, parent)
+                # append fixed node to tree and update ancestors lft/rgt
+                self.fixed_nodes[node.pk] = node
+                self.fixed_nodes[parent.pk].fixed_children.append(node.pk)
+                self.update_ancestors_rgt(node)
+
+    def calculate_lft_rgt(self, node: Page, parent: Page):
+        """
+        add a new node to the existing MPTT structure. As we sorted by lft, we always add
+        to the right of existing nodes.
+        """
+        if not parent.fixed_children:
+            # first child node, use lft of parent to calculate node lft/rgt
+            node.lft = parent.lft + 1
+            node.rgt = node.lft + 1
+            node.depth = parent.depth + 1
+        else:
+            # parent has fixed_children. Get right-most sibling and continue lft from there.
+            left_sibling = self.fixed_nodes[parent.fixed_children[-1]]
+            node.lft = left_sibling.rgt + 1
+            node.rgt = node.lft + 1
+            node.depth = left_sibling.depth
+        return node
+
+    def update_ancestors_rgt(self, node: Page):
+        """
+        As we only append siblings to the right, we only need to modify the rgt values
+        of all ancestors to adopt the new node into the tree.
+        """
+        parent = self.fixed_nodes[node.parent.pk]
+        self.fixed_nodes[parent.pk].rgt = node.rgt + 1
+        node = self.fixed_nodes[parent.pk]
+        while node.parent:
+            self.fixed_nodes[parent.pk].rgt = node.rgt + 1
+            node = node.parent
+
+    def get_fixed_tree_nodes(self):
+        """
+        Yield all page tree nodes
+        """
+        yield self.fixed_nodes.values()
```

Comment:
simply returning the `dict_values` object makes more sense, doesn't it? there are no values being yielded individually like this, which I guess was the intention
```suggestion
        return self.fixed_nodes.values()
```